tests/train.py

```diff
@@ -0,0 +1,258 @@
+                #value = party.get('value', [])
+                party['value'].append(1.10+idx)
+                #party['value'] = value
+                aggr.assign_value(party['id'], party['value'])
+
+        LOGGER.info(f"Participants: {participants}")
+
+        #Round 2 - specific participant
+        LOGGER.info("Training round 2")
+        self.aggr_start(aggr, participants[0]['id'])
+
+        self.worker(workers[0], 0, 1)
+
+        self.aggr_wait(aggr, 1, fflabc.Notification.participant_updated)
+        LOGGER.info("Aggregator consolidates worker updates")
+
+        with aggr:
+            party = participants[0]
+            #value = party.get('value', [])
+            party['value'].append(1.12)
+            #party['value'] = value
+            aggr.assign_value(party['id'], party['value'])
+
+        LOGGER.info(f"Participants: {participants}")
+
+        #Aggregator now closes the task
+        with aggr:
+            LOGGER.info("Aggregator stopping the task...")
+            aggr.stop_task({'the_model':'the final model'})
+            LOGGER.info("Aggregator successfully completes the task")
+
+        LOGGER.info("***** FL now complete *****")
+        LOGGER.info("***** Retrospectively recall model lineage *****")
+
+        LOGGER.info("--------------------------------------------------------")
+        LOGGER.info("User1 Model Lineage")
+
+        with self.user[0] as worker:
+            result = worker.model_lineage(self.task)
+            if not min_model:
+                self.assertEqual(len(result), 2)
+            self.print_lineage(result)
+
+        LOGGER.info("--------------------------------------------------------")
+        LOGGER.info("User2 Model Lineage")
+
+        with self.user[1] as worker:
+            result = worker.model_lineage(self.task)
+            if not min_model:
+                self.assertEqual(len(result), 1)
+            self.print_lineage(result)
+
+        LOGGER.info("--------------------------------------------------------")
+        LOGGER.info("Aggregator Full Model Lineage")
+        with aggr_user:
+            result = aggr_user.model_lineage(self.task)
+            if not min_model:
+                self.assertEqual(len(result), 6)
+            self.print_lineage(result)
+
+        #Should work because of ACL
+        with self.user[1] as worker:
+            result = worker.get_model(self.task)
+            self.assertEqual(result, {'the_model':'the final model'})
+
+            #Should fail because of ACL
+            with self.assertRaises(fflabc.ServerException):
+                result = worker.delete_model(self.task)
+
+    def aggr_wait(self, aggr, worker_id, wanted, timeout=0.1):
+        #Aggregator enters listening mode and waits for notifications
+        with aggr:
+            result = aggr.receive(timeout)
+            LOGGER.debug(f"Aggregator received: {result.content}")
+            #LOGGER.info(f"Aggregator received: {result.notification}")
```

Comment:
Perhaps clean up the commented LOGGER commands?

Reply:
Cool.